contracts/Migration/MantleTokenMigratorV2.sol

```diff
@@ -0,0 +1,409 @@
+
+        // transfer MNT tokens to user, if there are insufficient tokens, in the contract this will revert
+        ERC20(MNT_TOKEN_ADDRESS).safeTransfer(_to, _amount);
+
+        emit TokensMigrated(_to, _amount);
+    }
+
+    /* ========== ADMIN UTILS ========== */
+
+    // Ownership Functions
+
+    /// @notice Transfers ownership of the contract to a new address
+    /// @dev emits an {OwnershipTransferred} event
+    /// @dev Requirements:
+    ///     - The caller must be the contract owner
+    function transferOwnership(address _newOwner) public onlyOwner {
+        owner = _newOwner;
+
+        emit OwnershipTransferred(msg.sender, _newOwner);
+    }
+
+    // Role Assignment Functions
+
+    /// @notice Gives the Approver role to an address
+    /// @dev emits an {ApproverRoleAssigned} event
+    /// @dev Requirements:
+    ///     - The caller must be the contract owner
+    /// @param _approver The address to give the Approver role to
+    function giveApproverRole(address _approver) public onlyOwner {
+        approvers[_approver] = true;
+
+        emit ApproverRoleAssigned(_approver);
+    }
+
+    /// @notice Revokes the Approver role from an address
+    /// @dev emits an {ApproverRoleRevoked} event
+    /// @dev Requirements:
+    ///     - The caller must be the contract owner
+    /// @param _approver The address to revoke the Approver role from
+    function revokeApproverRole(address _approver) public onlyOwner {
+        approvers[_approver] = false;
+
+        emit ApproverRoleRevoked(_approver);
+    }
+
+    /// @notice Gives the Blacklister role to an address
+    /// @dev emits an {BlacklisterRoleAssigned} event
+    /// @dev Requirements:
+    ///     - The caller must be the contract owner
+    /// @param _blacklister The address to give the Blacklister role to
+    function giveBlacklisterRole(address _blacklister) public onlyOwner {
+        blacklisters[_blacklister] = true;
+
+        emit BlacklisterRoleAssigned(_blacklister);
+    }
+
+    /// @notice Revokes the Blacklister role from an address
+    /// @dev emits an {BlacklisterRoleRevoked} event
+    /// @dev Requirements:
+    ///     - The caller must be the contract owner
+    /// @param _blacklister The address to revoke the Blacklist role from
+    function revokeBlacklisterRole(address _blacklister) public onlyOwner {
+        blacklisters[_blacklister] = false;
+
+        emit BlacklisterRoleRevoked(_blacklister);
+    }
+
+    // Contract State Functions
+
+    /// @notice Blacklists an address from token migration
+    /// @dev emits an {AddressBlacklisted} event
+    /// @dev Requirements:
+    ///     - The caller must have the Blacklister role
+    /// @param _address The address to blacklist
+    function blacklistAddress(address _address) public onlyBlacklister {
+        blacklistedAddresses[_address] = true;
+
+        emit AddressBlacklisted(msg.sender, _address);
```

Comment:
we could recycle and use something like `BlacklistModified(address, address, bool)` to reduce the number of events (and indexer handlers) but strictly a suggestion.

Reply:
combined